assets/maximo.js

```diff
@@ -242,17 +243,83 @@ class Maximo {
     </IKO_ITEMMASTERSet>
     </SyncIKO_ITEMMASTER>`;
     
-        let response = await fetch('https://prod.manage.prod.iko.max-it-eam.com/maximo/api/os/IKO_ITEMMASTER?action=importfile', {
+        let response = await fetch('https://test.manage.test.iko.max-it-eam.com/maximo/api/os/IKO_ITEMMASTER?action=importfile', {
             method: "POST",
             headers: {
                 "filetype":"XML",
                 "apikey": this.login.userid,
+                //"preview": "1"
             },
             body: xmldoc,
         });
         let content = await response.json();
+        console.log(content);
         return parseInt(content.validdoc);
     }
+
+    async uploadImageToMaximo(image){
+        //check valid image type
+        if(image.type !== "image/jpeg"){
+            return 'fail';
+        }
+
+        //check valid item number        
+        let itemnum = image.name.slice(0,7);
+        let response = await fetch(`https://test.manage.test.iko.max-it-eam.com/maximo/api/os/mxitem?oslc.where=itemnum=${itemnum}`, {
+            method: "GET",
+            headers: {
+                "apikey": this.login.userid,
+            }
+        })
+        let content = await response.json();
+        if(content["rdfs:member"] == 0 || content['oslc:Error']){
+            return 'fail';
+        }
+
+        //get item id
+        let itemId = content["rdfs:member"][0]["rdf:resource"];
+        itemId = itemId.slice(38);
+        //console.log("item id " + itemId);
+
+        //check for existing image
+        response = await fetch(`https://test.manage.test.iko.max-it-eam.com/maximo/api/os/mxitem/${itemId}`,{
+            method: "GET",
+            headers: {
+                "apikey": this.login.userid,
+            }
+        })
+        content = await response.json();
+
+        //delete image if it exists
+        if(content["_imagelibref"]){
+            //console.log("image exists");
+            //delete existing image
+            response = await fetch(`https://test.manage.test.iko.max-it-eam.com/maximo/api/os/mxitem/${itemId}?action=system:deleteimage`, {
```

Comment:
should just show a warning for that item instead, doubtful anyone would check for existing images before hand